src/bin/genemichaels.rs

```diff
@@ -244,14 +345,127 @@ fn main() {
                 Ok(_) => { },
                 Err(e) => {
                     if !args.quiet {
-                        eprintln!("Error formatting {}: {:?}", &file.to_string_lossy(), e);
+                        print_error_text();
+                        eprintln!("formatting {}: {:?}", &file.to_string_lossy(), e);
                     }
                     failed = true;
                 },
             };
         }
+        eprintln!(
+            "\x1B[1;32m    Finished\x1B[0;22m workspace formatted successfully in {:.2}s",
+            time::Instant::now().duration_since(inst).as_secs_f64()
+        );
         if failed {
             process::exit(1);
         }
     }
 }
+
+fn process_cargo_toml(path: PathBuf, manifest: Manifest, args: &Args, config: FormatConfig) {
+    let dirs = match manifest.workspace.map(|wkspc| wkspc.members) {
+        Some(mut mb) => {
+            if let Some(bins) = manifest.bin {
+                for bin in bins {
+                    mb.push(
```

Comment:
I think it would be good to make a new Vec here.  You're restricted by the fact that `members` needs strings rather than paths, and reusing a field for another purpose than it was originally intended would confuse me if I were just jumping in.

Reply:
Good call.